code/game/objects/throwing.dm

```diff
@@ -129,6 +129,7 @@
 	var/interval = 10 / speed
 	var/sleep_debt = 0
 	//use a modified version of Bresenham's algorithm to get from the atom's current position to that of the target
+	src.base_speed = speed
```

Comment:
this isn't a good way to do things. Use extension statmods, movespeed multiplicative, work it into these calculations